src/openrct2/ride/Ride.cpp

```diff
@@ -7761,10 +7768,10 @@ uint64_t Ride::GetAvailableModes() const
     if (gCheatsShowAllOperatingModes)
         return AllRideModesAvailable;
 
-    return GetRideTypeDescriptor().RideModes;
+    return GetRideTypeDescriptor()->RideModes;
 }
 
-const RideTypeDescriptor& Ride::GetRideTypeDescriptor() const
+const RideTypeDescriptor* Ride::GetRideTypeDescriptor() const
 {
-    return RideTypeDescriptors[type];
+    return &RideTypeDescriptors[type];
```

Comment:
I find this change a bit weird. How can this be a `nullptr`? Do you not initialize the RTD array entirely?

Besides that, you're only checking for the return to be valid on a couple of places.

Reply:
Agreed this should return a `const RideTypeDescriptor& ` as there will be no situation where this is nullptr.

Reply:
There *will*, once ride types are turned into objects.

Reply:
Hmm, then what about returning an optional?

Reply:
I went for consistency with the other object entry functions (e.g. ride entry, footpath entry, wall entry, ...).